sequence/src/main/java/com/vitreoussoftware/bioinformatics/sequence/io/reader/BufferFileStreamReader.java

```diff
@@ -0,0 +1,228 @@
+     *
+     * @param flag The {@link String} to search for
+     */
+    public void dropUntil(String flag) {
+        final int maxPosition = flag.length();
+        int position = 0;
+
+        while (position != maxPosition) {
+            if (index >= length)
+                bufferData();
+            if (length == -1)
+                return;
+            if (buffer[index] == flag.charAt(position)) {
+                position++;
+            }
+            else {
+                position = 0;
+            }
+            index++;
+        }
+    }
+
+    /**
+     * Ignore characters in the buffer until the {@link Predicate} {@see predicate} evaluates to true
+     * @param predicate The {@link Predicate} to test
+     */
+    public void dropUntil(Predicate<Character> predicate) {
+        while (true) {
+            if (index >= length)
+                bufferData();
+            if (length == -1 || predicate.test(buffer[index]))
+                return;
+
+            index++;
+        }
+    }
+
+    /**
+     * Ignore characters in the buffer while they are whitespace
+     */
+    public void dropWhileWhitespace() {
+        while (true) {
+            if (index >= length)
+                bufferData();
+            if (length == -1 || !Character.isWhitespace(buffer[index]))
+                return;
+
+            index++;
+        }
+    }
+
+    /**
+     * Ignore characters in the buffer until they are whitespace
+     */
+    public void dropUntilWhiteSpace() {
+        while (true) {
+            if (index >= length)
+                bufferData();
+            if (length == -1 || Character.isWhitespace(buffer[index]))
+                return;
+
+            index++;
+        }
+    }
+
+    /**
+     * Ignore the next character in the buffer
+     * @return How many characters were dropped
+     */
+    public int drop() {
+        return drop(1);
+    }
+
+    /**
+     * Ignore the next {@see characters} characters in the buffer
+     * @param characters The number of characters to drop
+     * @return How many characters were dropped
+     */
+    public int drop(int characters) {
+        index += characters;
+
+        return characters;
+    }
+
+    /**
+     * Take characters from the buffer until the {@link Predicate} {@see predicate} is found [start, predicate)
+     * @param predicate The {@link Predicate} to test
+     *
+     * @return The characters taken from the buffer
+     */
+    public String takeUntil(Predicate<Character> predicate) {
+        StringBuilder sb = new StringBuilder();
+
+        boolean found = false;
+        while (!found) {
+            if (index >= length)
+                bufferData();
+            if (length == -1 || predicate.test(buffer[index]))
+                found = true;
+            else
+                sb.append(buffer[index]);
+            index++;
+        }
+
+        return sb.toString().trim();
```

Comment:
The fact that the resulting string is trimmed isn't evident from the comments on this function, although I imagine there's a reason for the result to be trimmed.